### PR TITLE
[2.26.x] Historian blacklist for updates and deletes

### PR DESCRIPTION
**Pull request by malmgrens4**
#### What does this PR do?
Introduces system properties for skipping historian versioning on deletes and updates (both direct metacard updates and associated content updates). 

e.g.
`# Skip historian for the following types on updates`
`org.codice.ddf.history.update.blacklist.metacardTypes=query,workspace`

`# Skip historian for the following types on deletes`
`org.codice.ddf.history.deletes.blacklist.metacardTypes=workspace`

Additionally this copies the DeleteRequest properties onto the DeleteResponse. This was initially done to allow for the SKIP_VERSIONING property to be used with local deletes, but this also makes behavior more consistent across the Create, Update, and Delete requests on the Solr implementation of the CatalogProvider.

#### What are the relevant tickets?
Fixes: #____

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java [hunk 1]
```diff
@@ -223,6 +240,7 @@ public UpdateStorageResponse version(
             .map(ContentItem::getMetacard)
             .filter(Objects::nonNull)
             .filter(isNotVersionNorDeleted)
+            .filter(this::isNotBlackListedUpdate)
```
Comment (malmgrens4) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java:243]: We could split the config to allow for versioning on the content separately but not sure there's a use case for skipping versioning on the direct metacard updates, but not the content updates.

**Comment by alexabird** [comment]
build now

**Comment by malmgrens4** [comment]
build now

**Review by pvargas — APPROVED** [review]

File: catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrCatalogProviderImpl.java [hunk 2]
```diff
@@ -363,7 +363,7 @@ public DeleteResponse delete(DeleteRequest deleteRequest) throws IngestException
       deleteListOfMetacards(deletedMetacards, identifierPaged, attributeName);
     }
 
-    return new DeleteResponseImpl(deleteRequest, null, deletedMetacards);
+    return new DeleteResponseImpl(deleteRequest, deleteRequest.getProperties(), deletedMetacards);
```
Comment (rzwiefel) [inline catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrCatalogProviderImpl.java:366]: I could be misremembering, but it's probably worth checking that there's nothing sensitive inside the request properties that we don't want to send in the response.

**Review by rzwiefel — APPROVED** [review]
Nice historian updates 👉😏👉

**Review by derekwilhelm — APPROVED** [review]

File: catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrCatalogProviderImpl.java (hunk 2 above)
Comment (kcwire) [reply catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrCatalogProviderImpl.java:366]: I would agree. I think we should leave the request properties where they are and check them if there is a property we're looking for (I just made an update to do that to check for the skip_versioning property https://github.com/codice/ddf/pull/6789).

Comment (malmgrens4) [reply catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrCatalogProviderImpl.java:366]: Yea makes sense. I'll close this one and make an update to yours

Comment (malmgrens4) [reply catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrCatalogProviderImpl.java:366]: Okay yea I see your change now. I'll drop that change in my PR and we can have these tested together.

**Comment by alexabird** [comment]
build now

**Pull request merged**